h2o-core/src/main/java/hex/faulttolerance/Recoverable.java

```diff
@@ -0,0 +1,26 @@
+package hex.faulttolerance;
+
+import water.Key;
+import water.Keyed;
+
+import java.util.Set;
+
+public interface Recoverable<T extends Keyed> {
+
+    /**
+     * @return key of this keyed object
+     */
+    Key<T> getKey();
+
+    /**
+     * @param location directory where this recoverable will be written into a single file
+     * @return path to where data was written
+     */
+    String exportBinary(String location);
```

Comment:
this is not part of public API so, for now I think we are good and since another target is autoML I think one file should be enough, we can alway change this if it is not sufficient